--- a/etc/init.d/dropbear
+++ b/etc/init.d/dropbear
@@ -118,18 +118,6 @@
 
 start_service()
 {
-	[ x"bluetooth" == x`micocfg_work_mode` ] && {
-        logger -t dropbear.init.d -p 3  "in bluetooth mode, will not start"
-        return
-    }
-
-	ssh_en=`cat /data/ssh_en 2>/dev/null`
-	ssh_en_bind=`cat /data/.ssh_en 2>/dev/null`
-	ssh_en_tmp=`cat /tmp/ssh_en 2>/dev/null`
-	channel=`micocfg_channel 2>/dev/null`
-	if [ "$ssh_en" != "1" -a "$ssh_en_bind" != "1" -a "$ssh_en_tmp" != "1" -a "$channel" = "release" ]; then
-		return 0                                                                                  
-	fi
 	[ -s /data/etc/dropbear/dropbear_rsa_host_key ] || keygen
 
 	. /lib/functions.sh
